src/a.tex

```diff
@@ -306,19 +369,18 @@ \section{Eventual Success of Store-Conditional Instructions}
 \itemsep 1pt
 \item {\em H} or some other hart executes a successful SC to the reservation
   set of the LR instruction in {\em H}'s constrained LR/SC loops.
-\item Some other hart executes an unconditional store or AMO instruction to
-  the reservation set of the LR instruction in {\em H}'s constrained LR/SC
-  loop, or some other device in the system writes to that reservation set.
+\item The LR instruction's reservation is invalidated by one of the possible
+  causes of invalidation listed earlier other than {\em H} executing LR or SC.
 \item {\em H} executes a branch or jump that exits the constrained LR/SC loop.
-\item {\em H} takes a trap.
+\item {\em H} traps.
 \end{itemize}
 
 \begin{commentary}
 As a consequence of this requirement, if some harts in an execution
-environment are executing constrained LR/SC loops, and no other harts or
-devices in the execution environment execute an unconditional store or AMO to
-that reservation set, then at least one hart will eventually exit its
-constrained LR/SC loop.
+environment are executing constrained LR/SC loops, and no harts or
+devices in the execution environment write to the reservation set
+(as described in causes of reservation invalidation),
+then at least one hart will eventually exit its constrained LR/SC loop.
```

Comment:
This isn't quite right, because a successful SC is itself a write, so some hart does write the reservation set.

Reply:
Good point.